oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/ttl/DataTTLKeeperTimer.java

```diff
@@ -63,6 +64,28 @@ public void start(ModuleManager moduleManager) {
         Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(
             new RunnableWithExceptionProtection(this::delete,
                 t -> logger.error("Remove data in background failure.", t)), 1, 5, TimeUnit.MINUTES);
+
+        Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(new RunnableWithExceptionProtection(this::deleteAndCreateOther, t -> logger.error("delete index in background failure.", t)), 1, 2, TimeUnit.HOURS);
+    }
+
+    private void deleteAndCreateOther() {
+        TimeBuckets timeBuckets = convertTimeBucket(new DateTime());
+        IModelGetter modelGetter = moduleManager.find(CoreModule.NAME).provider().getService(IModelGetter.class);
+        List<Model> models = modelGetter.getModels();
+        DownsamplingConfigService downsamplingConfigService = moduleManager.find(CoreModule.NAME).provider().getService(DownsamplingConfigService.class);
+        List<Model> allModels = new ArrayList<>();
+        models.forEach(model -> {
```

Comment:
Create so many indexes daily? Look like crazy. I think only segment index has performance concern.

Reply:
The index that is created on a daily basis is determined in the application.yml configuration file. Not all indexes are created that way

Reply:
That makes sense. But, I still think, you could provide this feature as a part of es storage implementation only. Not in the core.

Reply:
Well, yes, you may not have seen carefully the changes I made. The modifications in core are only historical data deletion operations. Specific changes in data storage and data search are all modified in the module storage-elasticsearch-plugin

Reply:
Delete is one of the concerns too. If that delete API changed, it means all storage implementors(such as H2, MySQL) need to understand the ES index things, which make the API change. But actually, they may don't need to know.

You could think in this way, if I say the index created by daily is another form of storage implementor, how do you do the changes?

The best way is, you could deal with the delete inside the existing delete method. 

This made my another concern, `query/TraceQueryService.java`. You changed query too, which is much harder to be understood by frontend developer and others do the integration.

If you could support all indexes running by daily,  it is cool, but consider this as an advanced feature. Or, we could say, a new `elasticsearch-large-scale` provider, based on the existing one.

Reply:
Oh, I totally understand what you mean. First of all, thanks for your patient explanation. I think I know what to do.

Reply:
Look forward your update :)

`elasticsearch-large-scale` looks like really reasonable to me. In there, you could change settings, and make them more reasonable and clear, also reuse the existing DAO implementation.

Reply:
Yes, I can make great progress in this way, and I am also willing to do such a thing :)